app-modules/campaign/src/Filament/Resources/CampaignResource/Pages/CreateCampaign.php

```diff
@@ -71,7 +73,18 @@ protected function getSteps(): array
                         ->required(),
                     Select::make('segment_id')
                         ->label('Population Segment')
-                        ->options($user->segments()->pluck('name', 'id'))
+                        ->options(function () use ($user) {
+                            $teamIds = $user->teams->pluck('id');
+                            $users = TeamUser::query()
+                                ->whereIn('team_id', $teamIds)
+                                ->pluck('user_id');
+                            $users->push($user->getKey());
+                            $users->unique();
+
+                            return Segment::query()
+                                ->whereIn('user_id', $users)
```

Comment:
I feel like this could be more efficient with a `whereHas()`, do you have an idea on how to approach that or should I write some example code?